include/proxy-wasm/wasm.h

```diff
@@ -290,7 +290,11 @@ class WasmHandleBase : public std::enable_shared_from_this<WasmHandleBase> {
     }
   }
 
-  void kill() { wasm_base_ = nullptr; }
+  void kill() {
+    wasm_base_->startShutdown();
+    wasm_base_->finishShutdown();
+    wasm_base_ = nullptr;
```

Comment:
Calling `finishShutdown()` immediatelly after `startShutdown()`, without waiting for the plugin to call `proxy_done` is technically incorrect, and effectlively a no-op. This sequence works only if the plugin returns `true` from the `proxy_on_done` callback, in which case you don't even need to call `finishShutdown()`.

It also breaks the promise of not executing anything in the canary (e.g. you can imagine plugins sending batch reports on shutdown, etc.), which is why we're simply dropping the VM right now.